### PR TITLE
Feature/sc 22788/dataset v2 0 crud

### PR DESCRIPTION
**Pull request by ctiml**
**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**
Feature

**What this PR does / why we need it**:
Dataset 2.0 CRUD

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

## Test Image
- ~~sc22788-65bca6c~~
- ~~sc22788-6f51637~~
- sc-22788-7e70bc

### CODE REVIEW

File: packages/client/src/containers/Datasets/DatasetList.tsx [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+            pathname: `datasets/${record.id}`,
+          }}
+        >
+          {text}
+        </Link>
+      </TruncateTableField>
+    );
+  }
+
+  function renderTags(text, record) {
```
Comment (neighborhood999) [inline packages/client/src/containers/Datasets/DatasetList.tsx:162]: Seems the text not be used, you can change it as:

```suggestion
  function renderTags(record) {
```

And don't forget to delete Line 275 (`dataIndex`).

**Review by popcornylu — CHANGES_REQUESTED** [review]
Need to add dataset to `main.model_deploy.tsx`

**Review by popcornylu — CHANGES_REQUESTED** [review]
1. If there is a file in the `datasets` folder, it would crash the graphql.
2. If there is a folder in the `datasets/xxxx folder but no `.dataset` file, the graphql would fail.

```
6 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"}
/usr/src/app/packages/graphql-server/lib/resolvers/datasetV2.js:94
            const id = obj.prefix.replace(objectPrefix, '').slice(0, -1);
                                  ^

TypeError: Cannot read property 'replace' of undefined
    at Readable.<anonymous> (/usr/src/app/packages/graphql-server/lib/resolvers/datasetV2.js:94:35)
    at Readable.emit (events.js:400:28)
    at addChunk (internal/streams/readable.js:293:12)
    at readableAddChunk (internal/streams/readable.js:267:9)
    at Readable.push (internal/streams/readable.js:206:10)
    at Readable.readStream._read (/usr/src/app/node_modules/minio/dist/main/minio.js:1825:22)
    at DestroyableTransform.<anonymous> (/usr/src/app/node_modules/minio/dist/main/minio.js:1842:22)
    at DestroyableTransform.emit (events.js:400:28)
    at addChunk (/usr/src/app/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:298:12)
    at readableAddChunk (/usr/src/app/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:280:11)
    at DestroyableTransform.Readable.push (/usr/src/app/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:241:10)
    at DestroyableTransform.Transform.push (/usr/src/app/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:139:32)
    at DestroyableTransform._flush (/usr/src/app/node_modules/minio/dist/main/transformers.js:96:14)
    at DestroyableTransform.prefinish (/usr/src/app/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:129:10)
    at DestroyableTransform.emit (events.js:400:28)
    at prefinish (/usr/src/app/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:611:14)
```

**Comment by neighborhood999** [comment]
Conflict

File: packages/client/src/ee/main.model_deploy.tsx [hunk 2]
```diff
@@ -24,6 +24,7 @@ import DeploymentEditPage from 'ee/containers/deploymentEditPage';
 import GroupSettingsPage from 'containers/groupSettingsPage';
 import GroupSettingsDeployments from 'ee/components/groupSettings/deployments';
 import GroupSettingsMLflow from 'ee/components/groupSettings/mlflow';
+import { DatasetList, DatasetDetail } from 'containers/Datasets';
```
Comment (neighborhood999) [inline packages/client/src/ee/main.model_deploy.tsx:27]: Should we align the import component by `import Component from '.../path'` this way?

File: packages/client/src/components/datasets/CreateForm.tsx [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    }
+  }, [dataset]);
+
+  React.useEffect(() => {
+    if (inputVisible) {
+      inputRef.current?.input.focus();
+    }
+  }, [inputVisible]);
+
+  function genId(value: string) {
```
Comment (neighborhood999) [inline packages/client/src/components/datasets/CreateForm.tsx:39]: This is an independent function, you can move it outside the function component.

File: packages/client/src/components/datasets/CreateForm.tsx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+            Dataset ID: {id}
+          </div>
+          <Button type='default' onClick={onClose}>
+            Cancel
+          </Button>
+          <Button
+            type='primary'
+            disabled={id.length === 0 || name.length === 0}
+            onClick={async () => {
+              await onSubmit({ id, name, tags });
```
Comment (neighborhood999) [inline packages/client/src/components/datasets/CreateForm.tsx:103]: Maybe add a `try/catch`?

File: packages/client/src/containers/Datasets/DatasetDetail.tsx [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+  GroupContextComponentProps,
+  withGroupContext,
+} from 'context/group';
+import { errorHandler } from 'utils/errorHandler';
+
+import { Dataset, InputVariables } from 'components/datasets/common';
+import { DatasetCreateForm } from 'components/datasets/CreateForm';
+import { DatasetQuery, UpdateDatasetMutation } from './dataset.graphql';
+
+type Props = {
```
Comment (neighborhood999) [inline packages/client/src/containers/Datasets/DatasetDetail.tsx:28]: Providing another option:

```ts
interface Props extends RouteComponentProps<{ datasetId: string }>, GroupContextComponentProps {
  // ...
}
```

File: packages/client/src/containers/Datasets/DatasetDetail.tsx [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+  const [tabKey, setTabKey] = React.useState('data');
+
+  if (getDataset.error) {
+    return <div>Failure to load dataset.</div>;
+  }
+
+  async function onSubmit(data: Partial<InputVariables>) {
+    const { refetch, variables } = getDataset;
+
+    await updateDataset({
```
Comment (neighborhood999) [inline packages/client/src/containers/Datasets/DatasetDetail.tsx:55]: Add `try/catch`?

File: packages/client/src/containers/Datasets/DatasetDetail.tsx [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+      matcher: /\/datasets/,
+      title: 'Datasets',
+      link: '/datasets',
+    },
+    {
+      key: 'detail',
+      matcher: /\/datasets\/([\w-_])+/,
+      title: `Dataset: ${dataset.name}`,
+      tips: 'View the detailed information.',
+      // TODO: add doc link
```
Comment (neighborhood999) [inline packages/client/src/containers/Datasets/DatasetDetail.tsx:87]: Don't forget to put the link here if `tipsLink` is ready xd

File: packages/client/src/containers/Datasets/DatasetDetail.tsx [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+      variables: {
+        where: {
+          id: match.params.datasetId,
+          groupName: groupContext.name,
+        },
+      },
+      onCompleted: (data: any) => {
+        const dataset = data.updateDatasetV2;
+        notification.success({
+          message: `Dataset '${dataset.name}' (${dataset.id}) has been updated.`,
```
Comment (neighborhood999) [inline packages/client/src/containers/Datasets/DatasetDetail.tsx:246]: How about this way?
```suggestion
          message: <>Dataset <b>{dataset.name} ({dataset.id})</b> has been updated.<>,
```

**Review by neighborhood999 — APPROVED** [review]

**Pull request merged**